src/Interpolation/Interpolate.cpp

```diff
@@ -404,13 +404,22 @@ void Interpolate_Iterate( real CData[], const int CSize[3], const int CStart[3],
 #              ifdef MHD
                Aux_Message( stderr, "B field: " );
                for (int v=0; v<NCOMP_MAG; v++)     Aux_Message( stderr, " [%d]=%14.7e", v, FMag[i][v] );
-               Aux_Message( stderr, "\n" );
+               Aux_Message( stderr, " Emag=%14.7e\n", Emag );
 #              endif
 
-               if ( !FailBeforeEoS )
-               {
-                  if ( FData_is_Prim )    Aux_Message( stderr, "Eint=%14.7e\n", Eint );
-                  else                    Aux_Message( stderr, "Pres=%14.7e\n", Pres );
+//             output additional information
+               if ( FData_is_Prim ) {
+//                output Eint only if it has been recalculated
+                  if ( Eint != NULL_REAL )   Aux_Message( stderr, "Eint=%14.7e\n", Eint );
+               }
+
+               else {
+                  const real CheckMinPres_No = false;
+                  const real Pres = Hydro_Con2Pres( Temp[DENS], Temp[MOMX], Temp[MOMY], Temp[MOMZ], Temp[ENGY], Temp+NCOMP_FLUID,
+                                                    CheckMinPres_No, NULL_REAL, Emag,
+                                                    EoS_DensEint2Pres_CPUPtr, EoS_GuessHTilde_CPUPtr, EoS_HTilde2Temp_CPUPtr,
+                                                    EoS_AuxArray_Flt, EoS_AuxArray_Int, h_EoS_Table, &Eint );
+                  Aux_Message( stderr, "Eint=%14.7e, Pres=%14.7e\n", Eint, Pres );
```

Comment:
In the case `FData_is_Prim = false`, `Eint` is not recalculated yet.

Reply:
It is also returned by `Hydro_Con2Pres()` from the last argument.